components/src/dynamo/common/multimodal/media_connector.py

```
@@ -0,0 +1,108 @@
+            *,
+            image_mode: str = "RGB",
+        ) -> Image.Image:
+            """Async image fetch via dynamo's ImageLoader with LRU cache."""
+            try:
+                img = await self._image_loader.load_image(image_url)
+                # Ensure correct mode (ImageLoader always returns RGB,
+                # but respect the caller's request)
+                if image_mode != "RGB" and img.mode != image_mode:
+                    img = img.convert(image_mode)
+                return img
+            except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)